src/test/scala/org/constellation/storage/transactions/PendingTransactionsMemPoolTest.scala

```diff
@@ -151,6 +153,66 @@ class PendingTransactionsMemPoolTest extends FreeSpec with IdiomaticMockito with
 
       ret shouldBe None
     }
+
+    "it should return a consecutive chain of transactions and all the test transactions" in {
+      val memPool = PendingTransactionsMemPool[IO](txChainService, rl)
+
+      val tx1 = manuallyCreateTransaction("a", "", 0, false)
+      val tx2 = manuallyCreateTransaction("a", tx1.transaction.hash, tx1.transaction.ordinal, true)
+      val tx3 = manuallyCreateTransaction("a", tx2.transaction.hash, tx2.transaction.ordinal, false)
+      val tx4 = manuallyCreateTransaction("a", tx3.transaction.hash, tx3.transaction.ordinal, false)
+      val tx5 = manuallyCreateTransaction("a", tx4.transaction.hash, tx4.transaction.ordinal, true)
+
+      //not sending tx4 to pending to break the chain
+      (memPool.put(tx1.hash, tx1) >>
+        memPool.put(tx2.hash, tx2) >>
+        memPool.put(tx3.hash, tx3) >>
+        memPool.put(tx5.hash, tx5)).unsafeRunSync
+
+      val result = memPool.pull(10).unsafeRunSync
+
+      result shouldBe List(tx1, tx2, tx3, tx5).some
```

Comment:
should it return `tx5` if `tx4` is missing and `tx5` references to `tx4` not `tx3`?

Reply:
Yes because tx5 is a test transaction and previous logic allowed all the test transactions in. So I'm not changing that. Also there is an assumption that test transactions can be a part of the chain - we discussed that but didn't come to a conclusion that it's that or the other way around, so I remember I said that I will assume that they can be a part of a chain.